===================================================================
RCS file: /cvsroot/squirrelmail/squirrelmail/src/read_body.php,v
retrieving revision 1.179.2.20
retrieving revision 1.179.2.21
diff -u -r1.179.2.20 -r1.179.2.21
--- squirrelmail/squirrelmail/src/read_body.php	2002/11/16 06:06:25	1.179.2.20
+++ squirrelmail/squirrelmail/src/read_body.php	2002/12/05 13:24:43	1.179.2.21
@@ -9,7 +9,7 @@
  * This file is used for reading the msgs array and displaying
  * the resulting emails in the right frame.
  *
- * $Id: squirrelmail-1.2.10-gentoo.diff,v 1.1 2002/12/15 13:44:00 aliz Exp $
+ * $Id: squirrelmail-1.2.10-gentoo.diff,v 1.1 2002/12/15 13:44:00 aliz Exp $
  */
 
 require_once('../src/validate.php');
@@ -393,10 +393,10 @@
 $delimiter = $_SESSION['delimiter'];
 
 if (isset($_GET['passed_id'])) {
-    $passed_id = $_GET['passed_id'];
+    $passed_id = (int) $_GET['passed_id'];
 }
 elseif (isset($_POST['passed_id'])) {
-    $passed_id = $_POST['passed_id'];
+    $passed_id = (int) $_POST['passed_id'];
 }
 
 if (isset($_GET['sendreceipt'])) {
@@ -404,25 +404,25 @@
 }
 
 if (isset($_GET['sort'])) {
-    $sort = $_GET['sort'];
+    $sort = (int) $_GET['sort'];
 }
 elseif (isset($_POST['sort'])) {
-    $sort = $_POST['sort'];
+    $sort = (int) $_POST['sort'];
 }
 if (isset($_GET['startMessage'])) {
-    $startMessage = $_GET['startMessage'];
+    $startMessage = (int) $_GET['startMessage'];
 }
 elseif (isset($_POST['startMessage'])) {
-    $startMessage = $_POST['startMessage'];
+    $startMessage = (int) $_POST['startMessage'];
 }
 if (isset($_GET['show_more'])) {
-    $show_more = $_GET['show_more'];
+    $show_more = (int) $_GET['show_more'];
 }
 if (isset($_GET['show_more_cc'])) {
-    $show_more_cc = $_GET['show_more_cc'];
+    $show_more_cc = (int) $_GET['show_more_cc'];
 }
 if (isset($_GET['show_more_bcc'])) {
-    $show_more_bcc = $_GET['show_more_bcc'];
+    $show_more_bcc = (int) $_GET['show_more_bcc'];
 }
 if (isset($_GET['mailbox'])) {
     $mailbox = $_GET['mailbox'];
@@ -437,7 +437,7 @@
     $what = $_GET['what'];
 }
 if (isset($_GET['view_hdr'])) {
-    $view_hdr = $_GET['view_hdr'];
+    $view_hdr = (int) $_GET['view_hdr'];
 }
 if (isset($_SESSION['server_sort_array'])) {
     $server_sort_array = $_SESSION['server_sort_array'];

